torch/utils/_sympy/value_ranges.py

```diff
@@ -251,59 +219,23 @@ def ge(cls, a, b):
         return cls.not_(cls.lt(a, b))
 
     @staticmethod
-    def not_(a):
-        a = ValueRanges.wrap(a)
-        assert a.is_bool
-        if a.is_singleton():
-            return ValueRanges.wrap(sympy.Not(a.lower))
-        return ValueRanges(sympy.false, sympy.true)
-
-    @staticmethod
-    def to_dtype(x, dtype: torch.dtype):
-        def is_bool(val):
-            return isinstance(val, bool) or (
-                hasattr(val, "is_Boolean") and val.is_Boolean
-            )
-
-        x = ValueRanges.wrap(x)
-        low, up = x.lower, x.upper
-        if is_bool(low):
-            assert is_bool(up)
-            if dtype.is_floating_point:
-                return ValueRanges(0.0, 1.0)
-            else:
-                return ValueRanges(0, 1)
-        return x
-
-    @staticmethod
-    def constant(value, dtype):
-        # NB: value is NOT a sympy expression, it's a constant!
-        assert isinstance(value, (int, float, bool))
-        # using nan makes subsequent computation throw, and for the purposes of optimization
-        # returning -math.inf - math.inf is equivalent to giving up
-        if math.isnan(value):
-            return ValueRanges.unknown()
-        return ValueRanges.wrap(value)
-
-    @staticmethod
-    def reciprocal(x):
-        x = ValueRanges.wrap(x)
-        if 0 in x:
-            return ValueRanges.unknown()
-        else:
-            return ValueRanges.decreasing_map(x, lambda y: 1 / y)
+    def add(a, b):
+        return ValueRanges.coordinatewise_increasing_map(a, b, operator.add)
 
     @staticmethod
-    def square(x):
-        return ValueRanges.convex_min_zero_map(x, lambda y: y * y)
+    def mul(a, b):
+        def safe_mul(a, b):
+            if a == 0:
+                return 0
+            elif b == 0:
+                return 0
+            return a * b
 
-    @staticmethod
-    def abs(x):
-        return ValueRanges.convex_min_zero_map(x, abs)
+        return ValueRanges.coordinatewise_monotone_map(a, b, safe_mul)
 
-    @staticmethod
-    def neg(x):
-        return ValueRanges.decreasing_map(x, operator.neg)
+    @classmethod
+    def div(cls, a, b):
+        return cls.truediv(a, b)
```

Comment:
Once again, I'm not sure this is kosher because it is inconsistent with Inductor cpp codegen

```
    @staticmethod
    def div(a, b):
        return f"{a} / {b}"
```

from codegen.cpp

Reply:
Note that this change was not introduced in this PR, but in a previous one. The relevant discussion is in https://github.com/pytorch/pytorch/pull/100547#discussion_r1184646605